crates/turborepo-ui/src/tui/app.rs

```diff
@@ -255,35 +429,236 @@ fn update(
             app.next();
         }
         Event::ScrollUp => {
-            app.scroll(Direction::Up);
+            app.has_user_scrolled = true;
+            app.tasks
+                .get_mut(&app.active_task())
+                .unwrap()
+                .scroll(Direction::Up)
+                .unwrap_or_default();
```

Comment:
These feel like they should belong in a function on `App`. This allows us to test the state change via unit tests without setting up a channel to send events over. Instead we can just call `app.scroll(Director::Up)` and inspect them.